Running test_typing file directly fails

```python
C:\Users\KIRILL-1\CLionProjects\cpython> ./python Lib/test/test_typing.py
Running Debug|x64 interpreter...
.............................F....................................................................................................
..........................................................................................................s.......................
..................................................................................................................................
..................................................................................................................................
.......................................................
======================================================================
FAIL: test_repr (__main__.AnyTests.test_repr)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\KIRILL-1\CLionProjects\cpython\Lib\test\test_typing.py", line 118, in test_repr
    self.assertEqual(
AssertionError: "<class '__main__.AnyTests.test_repr.<locals>.Sub'>" != "<class 'test.test_typing.AnyTests.test_repr.<locals>.Sub'
>"
- <class '__main__.AnyTests.test_repr.<locals>.Sub'>
?          ^^^  ^^
+ <class 'test.test_typing.AnyTests.test_repr.<locals>.Sub'>
?         +++++++++ ^^^  ^


----------------------------------------------------------------------
Ran 575 tests in 0.815s

FAILED (failures=1, skipped=1)
```
It's easy to fix, just need to replace `"<class 'test.test_typing.AnyTests.test_repr.<locals>.Sub'>"` to `f"<class '{__name__}.AnyTests.test_repr.<locals>.Sub'>"`
I'll soon send a PR

<!-- gh-linked-prs -->
### Linked PRs
* gh-104037
* gh-104039
<!-- /gh-linked-prs -->

--- a/Lib/test/test_typing.py
+++ b/Lib/test/test_typing.py
@@ -117,7 +117,7 @@ class AnyTests(BaseTestCase):
         class Sub(Any): pass
         self.assertEqual(
             repr(Sub),
-            "<class 'test.test_typing.AnyTests.test_repr.<locals>.Sub'>",
+            f"<class '{__name__}.AnyTests.test_repr.<locals>.Sub'>",
         )
 
     def test_errors(self):

